Source/WebCore/platform/graphics/SourceBrush.h

```diff
@@ -34,22 +34,15 @@ namespace WebCore {
 
 class SourceBrush {
 public:
-    struct Brush {
-        using LogicalGradient = SourceBrushLogicalGradient;
-
-        friend bool operator==(const Brush&, const Brush&);
-
-        using Variant = std::variant<LogicalGradient, Ref<Pattern>>;
-        Variant brush;
-    };
+    using OptionalPatternGradient = std::variant<std::monostate, SourceBrushLogicalGradient, Ref<Pattern>>;
 
     SourceBrush() = default;
-    WEBCORE_EXPORT SourceBrush(const Color&, std::optional<Brush>&& = std::nullopt);
+    WEBCORE_EXPORT SourceBrush(const Color&, OptionalPatternGradient&& = std::monostate { });
 
     const Color& color() const { return m_color; }
     void setColor(const Color color) { m_color = color; }
 
-    const std::optional<Brush>& brush() const { return m_brush; }
+    const OptionalPatternGradient& patternGradient() const { return m_brush; }
```

Comment:
This function is only used by the encoder. The rest of the code is either getting the gradient or the pattern out of `m_brush`. Maybe if we add something like this the code will be a little bit cleaner:
```
const Pattern* pattern() const;
const SourceBrushLogicalGradient* gradient() const;
```

Reply:
They are there, just below this line.